src/mlpack/methods/ann/layer/dropout.hpp

```diff
@@ -77,6 +77,24 @@ class DropoutType : public Layer<MatType>
    */
   void Forward(const MatType& input, MatType& output);
 
+  /**
+   * Implementation of the forward pass of the dropout layer.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   */
+  template<typename T = MatType, typename std::enable_if_t<arma::is_arma_type<T>::value, int> = 0>
+  void ForwardImpl(const T& input, T& output);
+
+  /**
+   * General implementation of the forward pass of the dropout layer.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   */
+  template<typename T = MatType, typename std::enable_if_t<!arma::is_arma_type<T>::value, int> = 0>
+  void ForwardImpl(const T& input, T& output);
```

Comment:
@MarkFischinger
Looks better, just use directly `MatType` instead of `T`.  It is better for readability because `T`  does not mean a lot.
Also, could you break the line? mlpack code base is usually under 80 lines of code.
Also, the syntax is correct, I would remove `int` and instead add a `*` after `>` so the  signature looks as follows:
```c++
typename std::enable_if_t<!arma::is_arma_type<T>::value>* = 0>
```
The reason for this is to have the same signature over all the code base, making it easier for everyone.